kliff/trainer/base_trainer.py

```diff
@@ -0,0 +1,636 @@
+            configuration_transform: Union[dict, None] = self.transform_manifest.get(
+                "configuration", None
+            )
+            property_transform: Union[list, None] = self.transform_manifest.get(
+                "property", None
+            )
+
+            if property_transform:
+                for property_to_transform in property_transform:
+                    property_name = property_to_transform.get("name", None)
+                    if not property_name:
+                        continue  # it is probably an empty propery
+                    transform_module_name = property_to_transform[property_name].get(
+                        "name", None
+                    )
+                    if not transform_module_name:
+                        raise TrainerError(
+                            "Property transform module name not provided."
+                        )
+                    property_transform_module = importlib.import_module(
+                        f"kliff.transforms.property_transforms"
+                    )
+                    property_module = getattr(
+                        property_transform_module, transform_module_name
+                    )
+                    property_module = property_module(
+                        proprty_key=property_name,
+                        **property_to_transform[property_name].get("kwargs", {}),
+                    )
+                    self.dataset = property_module(self.dataset)
+                    self.property_transforms.append(property_module)
+
+            if configuration_transform:
+                configuration_module_name: Union[str, None] = (
+                    configuration_transform.get("name", None)
+                )
+                if not configuration_module_name:
+                    logger.warning(
+                        "Configuration transform module name not provided."
+                        "Skipping configuration transform."
+                    )
+                else:
+                    configuration_transform_module = importlib.import_module(
+                        f"kliff.transforms.configuration_transforms"
+                    )
+                    configuration_module = getattr(
+                        configuration_transform_module, configuration_module_name
+                    )
+                    kwargs: Union[dict, None] = configuration_transform.get(
+                        "kwargs", None
+                    )
+                    if not kwargs:
+                        raise TrainerError(
+                            "Configuration transform module options not provided."
+                        )
+                    configuration_module = configuration_module(
+                        **kwargs, copy_to_config=True
+                    )
+
+                    for config in self.dataset.configs:
+                        _ = configuration_module(config, return_extended_state=True)
```

Comment:
Unlike property, here configuration transform is only initialized but the actual transformation does not happen. Where is it performed?

Reply:
I have gone back and forth on this one several times. the reason why config transform is not applied here is that for large datasets like MPTraj, you run of memory really fast. So you evaluate transforms at model train time. For smaller datasets it is actually more useful to transform the model and save the binary for re-use. I think best would be to add a keyword `lazy_eval` etc. But it is much simpler this way that every configuration would be evaluated at runtime.